kratos/containers/distributed_csr_matrix.h

```diff
@@ -88,7 +88,11 @@ class DistributedCsrMatrix final
         //compute the columns size
         TIndexType max_col_index = rSparseGraph.ComputeMaxGlobalColumnIndex();
         TIndexType tot_col_size = max_col_index+1;
-        mpColNumbering = Kratos::make_unique<DistributedNumbering<TIndexType>>(*mpComm, tot_col_size, mpComm->Size());
+            
+        if(tot_col_size == mpRowNumbering->Size()) //this ensures that diagonal blocks are square for square matrices
+            mpColNumbering = Kratos::make_unique<DistributedNumbering<TIndexType>>(*mpComm, mpRowNumbering->GetCpuBounds());
+        else
+            mpColNumbering = Kratos::make_unique<DistributedNumbering<TIndexType>>(*mpComm, tot_col_size, mpComm->Size());
```

Comment:
```suggestion
        //this ensures that diagonal blocks are square for square matrices
        if (tot_col_size == mpRowNumbering->Size()) {
            mpColNumbering = Kratos::make_unique<DistributedNumbering<TIndexType>>(*mpComm, mpRowNumbering->GetCpuBounds());
        } else {
            mpColNumbering = Kratos::make_unique<DistributedNumbering<TIndexType>>(*mpComm, tot_col_size, mpComm->Size());
        }
```
Minor style guide suggestion.

Reply:
Thanks @rubenzorrilla, my TOC was triggered